common/page.go

```diff
@@ -166,8 +168,20 @@ func (p *Page) didCrash() {
 	p.emit(EventPageCrash, p)
 }
 
-func (p *Page) evaluateOnNewDocument(source string) {
-	// TODO: implement
+func (p *Page) evaluateOnNewDocument(source string) error {
```

Comment:
This change could go to a single commit (maybe the first commit as it's the actual goal of the PR).